src/uct/ib/base/ib_iface.c

```diff
@@ -754,6 +784,8 @@ static int uct_ib_iface_dev_addr_is_reachable(uct_ib_iface_t *iface,
         !((params.pkey | iface->pkey) & UCT_IB_PKEY_MEMBERSHIP_MASK) ||
         /* PKEY values have to be equal */
         ((params.pkey ^ iface->pkey) & UCT_IB_PKEY_PARTITION_MASK)) {
+        uct_iface_populate_info_str_buf(is_reachable_params,
+                                        "none of the pkeys has full membership");
```

Comment:
let's split to 2 cases:
1. "both local and remote pkeys 0x%x have partial membership"
2. "local pkey 0x%x differs from remote pkey 0x%x"